### PR TITLE
multi-property matching and including cards from other sets in booster gen

### PR DESCRIPTION
**Pull request by brine**
+- Can define additional properties to match with for card pack generator
+- Can include cards from other set defs within specific card packs
+- Can define alternative property values to cards included from other sets

### CODE REVIEW

**Comment by brine** [comment]
if you include a card from a different set, and modify their custom property values, then those properties will carry over through the booster generator into the sealed deck editor, but WON'T carry through to the actual game table after the deck is loaded.

For example, if I include an Uncommon card from a set in my pack and change this property to Rare within the pack, then the pack generator will treat the card as Rare, and the sealed deck filters will see it as Rare.  Once I load the deck, the card will return to its original Uncommon value.

One of the side effects of this, is the proxy generator will generate a proxy image of the card in it's ALTERED (not original state).  The proxy image is carried over to the play table after I load the deck, as nothing is currently forcing it to regenerate the image with the original property values.

I dunno if its something that needs to be dealt with or not, open to input here.

**Comment by kellyelton** [comment]
My reaction when reviewing this pull request
image

File: octgnFX/Octgn.DataNew/Entities/Pick.cs [hunk 1]
```diff
@@ -20,47 +21,48 @@ public Pick(XmlReader reader)
...
-        public string Key { get; set; }
-        public string Value { get; set; }
+        public List<Tuple<string, string>> Properties { get; set; }
 
         public PackContent GetCards(Pack pack, Set set)
         {
             var ret = new PackContent();
+            var cardList = set.Cards.ToList();
 
-            if (Quantity < 0)
```
Comment (kellyelton) [inline octgnFX/Octgn.DataNew/Entities/Pick.cs]: image

File: octgnFX/Octgn.DataNew/GameSerializer.cs [hunk 2]
```diff
@@ -1229,8 +1241,13 @@ internal PackDefinition DeserializePack(IEnumerable<XElement> element)
                         var pick = new Pick();
                         var qtyAttr = e.Attributes().FirstOrDefault(x => x.Name.LocalName == "qty");
                         if (qtyAttr != null) pick.Quantity = qtyAttr.Value == "unlimited" ? -1 : int.Parse(qtyAttr.Value);
-                        pick.Key = e.Attribute("key").Value;
-                        pick.Value = e.Attribute("value").Value;
+                        var propertyList = new List<Tuple<string, string>>();
```
Comment (kellyelton) [inline octgnFX/Octgn.DataNew/GameSerializer.cs]: image

File: octgnFX/Octgn.DataNew/Entities/Pick.cs [hunk 3]
```diff
@@ -20,47 +21,48 @@ public Pick(XmlReader reader)
...
 
+            if (Quantity < 0)
+            {
+                ret.UnlimitedCards.AddRange(cardList);
+            }
+            else
+            {
                 for (var i = 0; i < Quantity; i++)
                 {
-                    var pick = list.RandomElement();
```
Comment (kellyelton) [inline octgnFX/Octgn.DataNew/Entities/Pick.cs]: image

**Pull request closed without merging**